tokio-console/src/config.rs

```diff
@@ -238,3 +313,67 @@ impl FromStr for RetainFor {
         }
     }
 }
+
+// === impl ColorToggles ===
+
+impl ColorToggles {
+    pub fn color_durations(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+
+    pub fn color_terminated(&self) -> bool {
+        self.color_durations.map(std::ops::Not::not).unwrap_or(true)
+    }
+}
+
+// === impl ColorToggles ===
+
+impl ConfigFile {
+    fn from_config() -> Option<Self> {
+        let mut base = dirs::config_dir();
+        if let Some(path) = base.as_mut() {
+            path.push("tokio-console/console.toml");
+        }
+        let base = base.and_then(|path| fs::read_to_string(path).ok());
+        let base_file: Option<ConfigFile> = base.and_then(|raw| toml::from_str(&raw).ok());
+
+        let current = fs::read_to_string("./console.toml").ok();
+        let current_file: Option<ConfigFile> = current.and_then(|raw| toml::from_str(&raw).ok());
+        merge_config_file(base_file, current_file)
+    }
+
+    fn into_view_options(self) -> ViewOptions {
+        ViewOptions {
+            no_colors: self.colors.as_ref().map(|config| Not::not(config.enabled)),
+            lang: self.charset.as_ref().map(|config| config.lang.to_string()),
+            ascii_only: self.charset.as_ref().map(|config| config.ascii_only),
+            truecolor: self.colors.as_ref().map(|config| config.truecolor),
+            palette: self.colors.as_ref().map(|config| config.palette),
+            toggles: ColorToggles {
+                color_durations: self
+                    .colors
+                    .as_ref()
+                    .map(|config| Not::not(config.enable.durations)),
+                color_terminated: self.colors.map(|config| Not::not(config.enable.terminated)),
+            },
+        }
+    }
+}
+
+fn merge_config_file(before: Option<ConfigFile>, after: Option<ConfigFile>) -> Option<ConfigFile> {
```

Comment:
nit, take it or leave it: again, this seems like it could be better off as a method on `ConfigFile`?